src/app/qgshandlebadlayers.cpp

```diff
@@ -509,3 +479,162 @@ int QgsHandleBadLayers::layerCount()
 {
   return mLayerList->rowCount();
 }
+
+QString QgsHandleBadLayers::findFile( const QString &fileName, const QString &basePath, int maxDepth, int driveMargin )
+{
+  int depth = 0;
+  QDir folder = QDir( basePath );
+  QString existingBase;
+  while ( !folder.exists() && folder.absolutePath().count( folder.separator() ) > driveMargin )
+  {
+    existingBase = folder.path();
+    if ( existingBase.contains( '/' ) )
```

Comment:
Is this windows compatible?

Reply:
You are right, I'll use QDir::separator() instead, just to be safe.

Reply:
Not sure if there's a risk that on Windows there will be `/` also. And if this can be avoided by using e.g. `absolutePath()`, would be worth a check (I don't have a windows system right now, so I can't sorry)